components/org.wso2.carbon.identity.conditional.auth.functions.user/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/user/IsMemberOfAnyOfGroupsFunctionImpl.java

```diff
@@ -109,8 +109,13 @@ private boolean isFederatedUserMemberOfAnyGroup(JsAuthenticatedUser user, List<S
     private String getGroupsClaimURI(JsAuthenticatedUser user) {
 
         String groupsClaimURI = getGroupsClaimURIByClaimMappings(user);
-        if (groupsClaimURI == null && user.getContext().getCurrentAuthenticator() != null &&
-                OPENIDCONNECT_AUTHENTICATOR_NAME.equals(user.getContext().getCurrentAuthenticator())) {
+        if ((groupsClaimURI == null && user.getContext().getCurrentAuthenticator() != null &&
+                OPENIDCONNECT_AUTHENTICATOR_NAME.equals(user.getContext().getCurrentAuthenticator())) ||
+                (user.getContext().getPreviousAuthenticatedIdPs().containsKey(user.getContext()
```

Comment:
Won't this return null at any situation?

Reply:
PreviousAuthenticatedIdPs can be null on initial authentication. Here, we come to `user.getContext().getPreviousAuthenticatedIdPs()` only if the `getCurrentAuthenticator()` is null. If a session exists, there should be previous authenticators. 

Nevertheless, the logic is refined through https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/196/commits/f74079c84b5a5e7b2194dda175b9e0311f7ad5a2